#Ansible Module Name -- rds_acc_mgmt.py
#Test Case : Revoke Account permissions from account without permissions
#Ansible Playbook created by - Abha H
---
- name: Revoke Account permissions from account without permissions
  hosts: localhost
  connection: local
  vars_files:
     - ../../../vars/alicloud-creds.yml
     - ../../vars/test-environment.yml
     - ../../vars/general-parameters.yml
   
  tasks:
   - block :
     - name: Scenario 1 - create account
       rds_acc_mgmt:
         acs_access_key: '{{ aliyun_acs_access_key }}'
         acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
         region: '{{ region }}'
         command: '{{ command }}'
         db_instance_id: '{{ instanceId }}'
         name: '{{ accName }}'
         password: '{{ accPwd }}'
         description: '{{ description }}'
         account_type: '{{ accountType }}'
       register: result_scn1
     - debug: var=result_scn1   

     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no

   - pause :
        seconds: 30

   - block:
     - name: Scenario 2 - create database
       rds:
        acs_access_key: '{{ aliyun_acs_access_key }}'
        acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
        region: '{{ region }}'
        command: '{{ command }}'
        instance_id: '{{ instanceId }}'
        db_name: '{{ dbName }}'
        db_description: '{{ dbDescription }}'
        character_set_name: '{{ characterSetName }}'
       register: result_scn2
     - debug: var=result_scn2
     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no

   - pause :
        seconds: 30
        
   - pause :
        seconds: 30

   - block :
     - name: Scenario 3 - Revoke Account permissions from account without permissions
       rds_acc_mgmt:
         acs_access_key: '{{ aliyun_acs_access_key }}'
         acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
         region: '{{ region }}'
         command: '{{ commandRevoke }}'
         db_instance_id: '{{ instanceId }}'
         account_name: '{{ accName }}'
         db_name: '{{ dbName }}' 
       register: result_scn3
     - debug: var=result_scn3   

     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no
     
   - pause :
        seconds: 30
        
   - block :
      - name: Scenario 4 - delete account
        rds_acc_mgmt:
          acs_access_key: '{{ aliyun_acs_access_key }}'
          acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
          region: '{{ region }}'
          command: '{{ commandDelete }}'
          db_instance_id: '{{ instanceId }}'
          name: '{{ accName }}'  
        register: result_scn4
      - debug: var=result_scn4

     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no

   - pause:
        seconds: 30

   - block:
     - name: Scenario 5 - delete database
       rds:
        acs_access_key: '{{ aliyun_acs_access_key }}'
        acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
        region: '{{ region }}'
        command: '{{ commandDelete }}'
        instance_id: '{{ instanceId }}'
        db_name: '{{ dbName }}'
       register: result_scn5
     - debug: var=result_scn5
     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no
      
     always:
      - assert:  
         that:
            - "'failed' not in result_scn1"
            - "'failed' not in result_scn2"
            - "'failed' in result_scn3"
            - "'The Specified db does not exist or DB status does not support.' in result_scn3.msg[0]['Error Message']"
            - "'failed' not in result_scn4"
            - "'failed' not in result_scn5"
         msg: 'Test pass FAILED !!!'
               
        